Docs/notification-service.md

```diff
@@ -0,0 +1,169 @@
+# Notification Service and Push
```

Comment:
I don't have permission to edit the wiki, and I wanted to make sure this document gets a quick review so I uploaded here.

I'll remove it once we add it to the wiki and before landing this PR

Reply:
Added https://github.com/mozilla-mobile/firefox-ios/wiki/Notification-Service-and-Push